tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -159,140 +180,205 @@ public static void main(String[] args) throws Exception {
             if (!shouldPrintMetrics) {
                 producer.close();
 
-                /* print final results */
+            /* print final results */
                 stats.printTotal();
             } else {
                 // Make sure all messages are sent before printing out the stats and the metrics
                 // We need to do this in a different branch for now since tests/kafkatest/sanity_checks/test_performance_services.py
                 // expects this class to work with older versions of the client jar that don't support flush().
                 producer.flush();
 
-                /* print final results */
+            /* print final results */
                 stats.printTotal();
 
-                /* print out metrics */
+            /* print out metrics */
                 ToolsUtils.printMetrics(producer.metrics());
                 producer.close();
             }
         } catch (ArgumentParserException e) {
-            if (args.length == 0) {
-                parser.printHelp();
-                Exit.exit(0);
-            } else {
-                parser.handleError(e);
-                Exit.exit(1);
-            }
-        }
 
+            opts.parser.handleError(e);
+            Exit.exit(1);
```

Comment:
Are these above two lines would be somewhat equivalent with printUsageAndDie? What is the difference?

Reply:
It's more like `printErrorAndDie`. I'm going to change with this. Thanks !